packages/fxa-settings/src/components/InputPhoneNumber/index.tsx

```diff
@@ -0,0 +1,136 @@
+    validationPattern: phoneValidationNorthAmerica,
+  },
+  {
+    id: 2,
+    code: '+1',
+    classNameFlag: 'bg-flag-canada',
+    name: 'Canada',
+    ftlId: 'input-phone-number-country-canada',
+    validationPattern: phoneValidationNorthAmerica,
+  },
+];
+
+const InputPhoneNumber = ({
+  countries = defaultCountries,
+  register,
+}: {
+  countries?: Country[];
+  register: UseFormMethods['register'];
+}) => {
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedCountries = countries.map((country) => ({
+    ...country,
+    localizedName: ftlMsgResolver.getMsg(country.ftlId, country.name),
+  }));
+  const [selectedCountry, setSelectedCountry] = useState(localizedCountries[0]);
+
+  const localizedLabel = ftlMsgResolver.getMsg(
+    'input-phone-number-enter-number',
+    'Enter phone number'
+  );
+
+  const handleCountryChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selected = localizedCountries.find(
+      (country) => country.id === parseInt(event.target.value, 10)
+    );
+    if (selected) {
+      setSelectedCountry(selected);
+    }
+  };
+
+  return (
+    <div className="flex">
+      <select
+        onChange={handleCountryChange}
+        value={selectedCountry.code}
+        className={`bg-transparent border border-grey-200 rounded-md py-2 ps-10 w-[60px] me-2 focus:border-blue-400 focus:outline-none focus:shadow-input-blue-focus ${selectedCountry.classNameFlag} bg-no-repeat bg-[length:1.5rem_1rem] bg-[40%_50%]`}
+      >
+        {/* Selected country is always first */}
+        <option className={`${selectedCountry.classNameFlag} bg-contain`}>
+          {selectedCountry.localizedName} ({selectedCountry.code})
+        </option>
+
+        {/* Note, at the time of writing we're using react-dom 18.3 which complains about
+        <hr> inside <select> being invalid, but it is valid. This should be fixed in later
+         versions: https://github.com/facebook/react/issues/27572 */}
+        <hr className="my-1" />
+
+        {localizedCountries
+          .filter((country) => country.id !== selectedCountry.id)
+          .map((country) => (
+            <option
+              key={country.id}
+              value={country.id}
+              className={`${country.classNameFlag} bg-contain`}
+            >
+              {country.localizedName} ({country.code})
+            </option>
+          ))}
+      </select>
+
+      {/* Because the country code may not be unique, the above `select`'s `value` must
+       be by country ID. This hidden input allows us to access it in the form data. */}
+      <input
+        type="hidden"
+        name="countryCode"
+        value={selectedCountry.code}
+        ref={register}
+      />
+
+      {/* Using `type="text" inputmode="numeric"` shows the numeric keyboard on mobile
+      and strips out whitespace on desktop, but does not add an incrementer. */}
+      <InputText
+        name="phoneNumber"
+        type="text"
```

Comment:
perhaps we can use `type='tel'` to display the phone number keyboard on mobile - it defaults to `text` type if a browser doesn't support `tel`